site/en/blog/prerender-pages/index.md

```diff
@@ -0,0 +1,263 @@
+{% endAside %}
+
+## Impact on analytics
+
+Analytics is used to measure website usage, for example using Google Analytics to measure page views, and events. Or by measuring performance metrics of pages using [Real User Monitoring (RUM)](https://en.wikipedia.org/wiki/Real_user_monitoring).
+
+Pages should only be prerendered when there is a high probability the page will be loaded by the user. This is why the Chrome Omnibox prerendering options only happen when there is such a high probability (80% or higher).
+
+However, particularly when using the Speculation rules API, prerendered pages may have an impact on analytics and site owners may wish to add extra code to only enable analytics for prerendered pages on activation.
+
+This could be achieved by using a `Promise` which waits for the `prerenderingchange` event if a document is prerendering, or resolves immediately if it is now:
+
+```js
+// Set up a promise for when the page is activated,
+// which is needed for prerendered pages.
+const whenActivated = new Promise((resolve) => {
+  if (document.prerendering) {
+    document.addEventListener('prerenderingchange', resolve);
+  } else {
+    resolve();
+  }
+});
+
+async function initAnalytics() {
+  await whenActivated;
+  gtag('config', 'UA-12345678-1');
+}
+
+initAnalytics();
+```
+
+This is the method we have implemented on this site and on [https://web.dev](https://web.dev).
+
+### Measuring performance
+
+For measuring performance metrics, analytics should consider whether it is better to measure these based upon the activation time rather than the page load time that browser APIs will report.
+
+For Core Web Vitals, measured by Chrome through the [Chrome User Experience Report](/docs/crux/), these are intended to measure the user experience. So these are measured based on activation time. This will often result in a 0 second LCP for example, showing this is great way of improving your Core Web Vitals.
+
+From version 3.0.4, the [web-vitals.js library](https://github.com/GoogleChrome/web-vitals) has been updated to handle prerendered navigations in the same way Chrome measure Core Web Vitals. This version also flags prerendered navigations for those metrics in the [Metric.navigationType](https://github.com/GoogleChrome/web-vitals#metric) attribute, which is discussed next.
+
+### Measuring prerenders
+
+Whether a page is prerendered can be seen with a non-zero activationStart entry of [PerformanceNavigationTiming](https://developer.mozilla.org/docs/Web/API/PerformanceNavigationTiming), supplemented with changes for some navigation currently not reported by that. This can then be logged using a Custom Dimension, or similar when logging the page views:
```

Comment:
```suggestion
Whether a page is prerendered can be seen with a non-zero `activationStart` entry of [`PerformanceNavigationTiming`](https://developer.mozilla.org/docs/Web/API/PerformanceNavigationTiming), supplemented with changes for some navigation currently not reported by that. This can then be logged using a Custom Dimension, or similar when logging the page views:
```